pkg/query-service/app/http_handler.go

```diff
@@ -2575,20 +2576,25 @@ func (aH *APIHandler) RegisterMessagingQueuesRoutes(router *mux.Router, am *Auth
 
 	spanEvaluation.HandleFunc("/evaluation", am.ViewAccess(aH.getProducerConsumerEval)).Methods(http.MethodPost)
 
-	// -------------------------------------------------
-	// Celery-specific routes
-	celeryRouter := messagingQueuesRouter.PathPrefix("/celery").Subrouter()
+	// for other messaging queues, add SubRouters here
+}
 
-	// Celery overview routes
-	celeryRouter.HandleFunc("/overview", am.ViewAccess(aH.getCeleryOverview)).Methods(http.MethodPost)
+// RegisterThirdPartyApiRoutes adds third-party-api integration routes
+func (aH *APIHandler) RegisterThirdPartyApiRoutes(router *mux.Router, am *AuthMiddleware) {
 
-	// Celery tasks routes
-	celeryRouter.HandleFunc("/tasks", am.ViewAccess(aH.getCeleryTasks)).Methods(http.MethodPost)
+	// Main messaging queues router
+	thirdPartyApiRouter := router.PathPrefix("/api/v1/third-party-apis").Subrouter()
 
-	// Celery performance routes
-	celeryRouter.HandleFunc("/performance", am.ViewAccess(aH.getCeleryPerformance)).Methods(http.MethodPost)
+	// Domain Overview route
+	overviewRouter := thirdPartyApiRouter.PathPrefix("/overview").Subrouter()
 
-	// for other messaging queues, add SubRouters here
+	overviewRouter.HandleFunc("/list", am.ViewAccess(aH.getDomainList)).Methods(http.MethodPost)
+	overviewRouter.HandleFunc("/domain", am.ViewAccess(aH.getDomainInfo)).Methods(http.MethodPost)
+
+	// Fetchers
+	fetchRouter := thirdPartyApiRouter.PathPrefix("/overview").Subrouter()
```

Comment:
why can't we use overviewRouter ?

Reply:
this was supposed to be implemented, cleaned it up.